### PR TITLE
fix(dialog) make align attribute into an input of dialog actions directive instead of css

### LINKED ISSUE #18479: Clarify use of non-standard "align" attribute for mat-dialog-actions
#### Documentation Feedback

Documentation suggests using the `align` attribute to align the content of `mat-dialog-actions` directives to the middle or right. However, `align` is an obsolete HTML attribute and not valid HTML5, and my IDE's static code analysis tools rightfully complain about using it. If using `align` really is the right way of doing this regardless of W3C standards, this should be clarified in the docs (and possibly be explicitly declared as a property of the `MatDialogActions` directive to make IDEs stop complaining). 

**Affected documentation page:** https://material.angular.io/components/dialog/overview

### PR DESCRIPTION
**Pull request by PooSham**
Fixes multiple issues, such as self-documentation of the align attribute, type checking, better bindings, and IDE support. Because align is not standard for HTMLDivElement, it doesn't make much sense to assume end users to know they can use the align attribute.

Fixes #18479

### CODE REVIEW

**Comment by PooSham** [comment]
I can't figure out how to add target label, I assume I don't have the rights to do it. It should probably be a patch.

**Review by crisbeto — COMMENTED** [review]
I believe that this will be a breaking change for people using `[attr.align]="someExpression"`. Also it looks like a bunch of unrelated code was modified.

**Comment by PooSham** [comment]
> I believe that this will be a breaking change for people using `[attr.align]="someExpression"`.

Not for the regular , I kept the css for the align attribute so that this would still work. I thought I'd break it at first, but I figured it out. 

> Also it looks like a bunch of unrelated code was modified.

I modified moth

**Comment by PooSham** [comment]
@crisbeto 

I changed both in material-experimental and the regular material, including tests and example files. The formatting changed in some files as well automatically by clang-format (which is a recommended extension to VS Code). If you want, I can revert those changes.

Regarding the `[attr.align]="someExpression"` breaking, that is true for material-experimental, but not for the regular material. I will fix the issue in material-experimental. I do believe it should be broken at one point since align isn't a standard HTML attribute, but I guess that's not up to me to decide.

**Comment by PooSham** [comment]
@crisbeto @jelbourn Over a year in the making, will this be merged? :)

**Comment by PooSham** [comment]
I had configured a new email account in git, but it wasn't added to the CLA. It should be fixed now, could a repository manager please rerun the check? @crisbeto @jelbourn

**Comment by PooSham** [comment]
@crisbeto Can you please rerun the checks now that I've updated the CLA?

**Comment by crisbeto** [comment]
There's a bug in CircleCI that sometimes prevents the workflows from running and we don't have a way of triggering them. The one above which says that it needs approval is for deploying the dev app and it is optional.

**Comment by PooSham** [comment]
> There's a bug in CircleCI that sometimes prevents the workflows from running and we don't have a way of triggering them. The one above which says that it needs approval is for deploying the dev app and it is optional.

Oh ok, seems to have run now. Do you have any feedback on the PR? Is there anything that needs to be fixed?

File: src/material/dialog/dialog-content-directives.ts [hunk 1]
```diff
@@ -145,9 +145,15 @@ export class MatDialogContent {}
...
-  host: {'class': 'mat-dialog-actions'},
+  host: {
+    'class': 'mat-dialog-actions',
+    '[class.mat-dialog-actions-align-center]': 'align === "center"',
+    '[class.mat-dialog-actions-align-end]': 'align === "end"',
+  },
 })
-export class MatDialogActions {}
+export class MatDialogActions {
+  @Input() align?: 'center' | 'end' = undefined;
```
Comment (crisbeto) [inline src/material/dialog/dialog-content-directives.ts:155]: Rather than `undefined`, should it support `start` which doesn't apply any styling?

**Review by crisbeto — COMMENTED** [review]
It looks like CircleCI still didn't run. It might be easier to resubmit the PR.

File: src/material/dialog/dialog-content-directives.ts (hunk 1 above)
Comment (PooSham) [reply src/material/dialog/dialog-content-directives.ts:155]: Ok, I will resubmit a new PR.

Yes, that might be better for API consistency. I will do that change in my next PR

**Comment by PooSham** [comment]
New PR [here](https://github.com/angular/components/pull/24178). Hope it works this time...

@crisbeto

**Pull request closed without merging**